SQL does not support expressions with semicolons

If SQL query is followed by a semicolon ( ; ) - SQL engine would fail with the following error message:

```
Caused by: com.hazelcast.sql.HazelcastSqlException: Encountered ";" at line 1, column 41.
Was expecting one of:
    <EOF> 
    "AS" ...
    "EXCEPT" ...
    "EXTEND" ...
    "FETCH" ...
    "FOR" ...
    "GROUP" ...
    ...
```

**Expected behavior**
Even though we don't support multi queries - we should at least allow using semicolon in the end of an expression.

--- a/hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/QueryParser.java
+++ b/hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/QueryParser.java
@@ -26,6 +26,7 @@ import com.hazelcast.sql.impl.calcite.validate.HazelcastSqlValidator;
 import com.hazelcast.sql.impl.calcite.validate.types.HazelcastTypeFactory;
 import org.apache.calcite.prepare.Prepare.CatalogReader;
 import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
 import org.apache.calcite.sql.parser.SqlParseException;
 import org.apache.calcite.sql.parser.SqlParser;
 import org.apache.calcite.sql.parser.SqlParser.Config;
@@ -89,7 +90,11 @@ public class QueryParser {
     private QueryParseResult parse(String sql, SqlBackend sqlBackend) throws SqlParseException {
         Config config = createConfig(sqlBackend.parserFactory());
         SqlParser parser = SqlParser.create(sql, config);
-        SqlNode topNode = parser.parseStmt();
+        SqlNodeList statements = parser.parseStmtList();
+        if (statements.size() != 1) {
+            throw QueryException.error(SqlErrorCode.PARSING, "The command must contain a single statement");
+        }
+        SqlNode topNode = statements.get(0);
 
         HazelcastSqlValidator validator =
                 (HazelcastSqlValidator) sqlBackend.validator(catalogReader, typeFactory, conformance, arguments);

--- a/hazelcast-sql/src/test/java/com/hazelcast/sql/impl/calcite/parse/QueryParserTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/sql/impl/calcite/parse/QueryParserTest.java
@@ -20,7 +20,7 @@ import com.hazelcast.sql.impl.QueryException;
 import com.hazelcast.sql.impl.calcite.SqlBackend;
 import com.hazelcast.sql.impl.calcite.validate.HazelcastSqlValidator;
 import com.hazelcast.sql.impl.calcite.validate.types.HazelcastTypeFactory;
-import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
 import org.apache.calcite.prepare.Prepare.CatalogReader;
@@ -36,6 +36,7 @@ import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
 
 import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.fail;
 import static org.mockito.ArgumentMatchers.any;
@@ -45,7 +46,7 @@ import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyNoMoreInteractions;
 
-@RunWith(HazelcastParallelClassRunner.class)
+@RunWith(HazelcastSerialClassRunner.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
 public class QueryParserTest {
 
@@ -132,7 +133,7 @@ public class QueryParserTest {
         verify(sqlBackend, never()).unsupportedOperationVisitor(any());
     }
 
-    @Test(expected = QueryException.class)
+    @Test
     public void when_neitherImdgOrJetCanHandleSql_then_throwsException() {
         // given
         given(sqlValidator.validate(isA(SqlNode.class))).willThrow(new CalciteException("expected test exception", null));
@@ -140,6 +141,23 @@ public class QueryParserTest {
 
         // when
         // then
-        parser.parse("SELECT * FROM t");
+        assertThatThrownBy(() -> parser.parse("SELECT * FROM t"))
+                .isInstanceOf(QueryException.class);
+    }
+
+    @Test
+    public void test_trailingSemicolon() {
+        given(sqlValidator.validate(isA(SqlNode.class))).willReturn(validatedNode);
+        given(sqlBackend.unsupportedOperationVisitor(catalogReader)).willReturn(unsupportedOperatorVisitor);
+
+        parser.parse("SELECT * FROM t;");
+        parser.parse("SELECT * FROM t;;");
+    }
+
+    @Test
+    public void when_multipleStatements_then_fails() {
+        assertThatThrownBy(() -> parser.parse("SELECT * FROM t; SELECT * FROM t"))
+                .isInstanceOf(QueryException.class)
+                .hasMessage("The command must contain a single statement");
     }
 }

